File: Studio/src/app/data/entity/Location.ts
import { Guid } from 'guid-typescript';
import { BaseEntity } from './BaseEntity';
import { City } from './City';

export interface Location extends BaseEntity{
    locationName: string; // LocationName in C#
    cityId: Guid; // Guid represented as string
    city?: City; // Reference to City class, optional
    events?: Event[];

    
}

File: Studio-BE/src/NM.Studio.Domain/Entities/Locations/Location.cs
using NM.Studio.Domain.Entities.Bases;
using NM.Studio.Domain.Entities.Events;
using System.ComponentModel.DataAnnotations.Schema;

namespace NM.Studio.Domain.Entities.Locations
{
    [Table("Location")]
    public class Location : BaseEntity
    {
        [Column(TypeName = "nvarchar(255)")]
        public string LocationName { get; set; }
        
        [ForeignKey("City")]
        public Guid CityId { get; set; }
        
        public virtual City City { get; set; }
        public virtual IList<Event> Events { get; set; }
    }
}
